manager/manager.go

```diff
@@ -386,10 +544,12 @@ func (m *Manager) issue(ctx context.Context, volumeID string) error {
 
 		switch readyCondition.Reason {
 		case cmapi.CertificateRequestReasonIssued:
+			log.V(4).Info("CertificateRequest has been issued!")
 			break
 		case cmapi.CertificateRequestReasonFailed:
 			return false, fmt.Errorf("request %q has failed: %s", updatedReq.Name, readyCondition.Message)
```

Comment:
Do we consider call `m.deletePendingRequestPrivateKey(req.UID)` when the `CertificateRequest` is in `Failed` condition ? It likely to fail again with the same private key. Create a new CR might help resolving the problem if it is due to key reuse issue.

Reply:
We'll automatically use a new private key the next time anyway - if a request is failed, it is terminal so will never be returned again by `findPendingRequest` (i.e. it won't be re-used). This will in turn trigger a new CR to be created and new private key generated (or at least, a new call to generatePrivateKey).

The item will be deleted from the map once the CR is deleted, although yep perhaps a future optimisation could be to delete terminal failed items from the map a bit early just to save on memory.. but it shouldn't have any functional difference :)